### PR TITLE
add pyright dep and start typechecking.  fix a lot of the errors, and ignore swaths of others

### PR DESCRIPTION
**Pull request by toshok**
Turns out python3 type hints are just that: hints.  they aren't checked at all outside code that actually looks at them (`pydantic`, etc.)

Add a dev dep on `pyright` and a script we can run using `rye run typecheck`

### CODE REVIEW

**Comment by Domiii** [comment]
Why `pyright` instead of `pydantic` or others?

File: pyproject.toml [hunk 1]
```diff
@@ -6,36 +6,36 @@ authors = [
...
     "pathspec>=0.12.1",
     "pyaudio>=0.2.14",
-    "speechrecognition>=3.10.4",
-    "google-auth>=2.32.0",
-    "google-cloud-speech>=2.26.1",
-    "pytest>=8.3.1",
+    "pylint>=3.2.5",
     "pynput>=1.7.7",
-    "networkx>=3.3",
-    "colorlog>=6.8.2",
```
Comment (Domiii) [inline pyproject.toml:36]: Curious: How did you get rid of unused dependencies? Is there a `rye` command for that?

**Review by Domiii — APPROVED** [review]

**Comment by toshok** [comment]
> Why `pyright` instead of `pydantic` or others?

as far as I can tell, `pydantic` isn't a static type checker - it verifies types at runtime, but only when using pydantic types (like `BaseModel`, etc.)  pyright will check types for vanilla python.  `mypy` is an alternative, but it didn't seem happy in this repo/with our code, and I read some github issue comments saying pyright had more/better support.

File: pyproject.toml (hunk 1 above)
Comment (toshok) [reply pyproject.toml:36]: oh I meant to call that out - I didn't remove anything.  I reordered them alphabetically.

**Pull request merged**